homeassistant/components/alphaess/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for AlphaEss integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import aiohttp
+from alphaess import alphaess
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    client = alphaess.alphaess()
+
+    try:
+        await client.authenticate(data["username"], data["password"])
+
+    except aiohttp.ClientResponseError as e:
+        if e.status == 401:
+            raise InvalidAuth
+        else:
+            raise UnknownError
```

Comment:
```suggestion
            raise UnknownError from e
```

It would be good to understand what the error is. You can also just `raise e` to reraise the original exception.

Reply:
Thank you - that is a better result.